tests/pds/dataservice_test.go

```diff
@@ -627,6 +628,88 @@ var _ = Describe("{UpgradeDataServiceImage}", func() {
+				for _, node := range nodes {
+					log.InfoD("Node Name is: %s", node.Name)
+				}
+				var nodeName string
+				Step("Delete Pods and Cordon a node", func() {
+					nodeName = nodes[0].Name // Selecting the 1st node in the list to cordon
+					podsList, err := pdslib.GetPodsOfSsByNode(*deployment.ClusterResourceName, nodeName, namespace)
+					log.FailOnError(err, "Pod not found on this Node")
+					log.InfoD("Pods found on %v node. Trying to Cordon and Drain pods from this node now.", nodeName)
+					err = k8sCore.DrainPodsFromNode(nodeName, podsList, timeOut, maxtimeInterval)
+					log.FailOnError(err, "draining pod from the node failed")
+				})
+				Step("Validate Data Service Again", func() {
+					resourceTemp, storageOp, config, err := pdslib.ValidateDataServiceVolumes(deployment, ds.Name, dataServiceDefaultResourceTemplateID, storageTemplateID, namespace)
+					log.FailOnError(err, "error on ValidateDataServiceVolumes method")
+					ValidateDeployments(resourceTemp, storageOp, config, ds.Replicas, dataServiceVersionBuildMap)
+				})
+				Step("Validate no pods are on the cordoned node anymore", func() {
+					nodes, err := pdslib.GetNodesOfSS(*deployment.ClusterResourceName, namespace)
+					log.FailOnError(err, "Cannot fetch nodes of the running Data Service")
+					for _, nodeObj := range nodes {
+						log.InfoD("Node Name is: %s", nodeObj.Name)
+						if nodeObj.Name == nodeName {
+							log.FailOnError(errors.New("New Pod came up on the node that was cordoned."), "Unexpected error")
+						}
+					}
+					log.InfoD("The pods of the Stateful Set %v are not on the cordoned node. Moving ahead now.", *deployment.ClusterResourceName)
+				})
+				Step("Running Workloads before scaling up of dataservices ", func() {
+					log.InfoD("Running Workloads on DataService %v ", ds.Name)
+					var params pdslib.WorkloadGenerationParams
+					pod, dep, err = RunWorkloads(params, ds, deployment, namespace)
+					log.FailOnError(err, fmt.Sprintf("Error while genearating workloads for dataservice [%s]", ds.Name))
+				})
+				Step("Delete the workload generating deployments", func() {
+					if Contains(dataServiceDeploymentWorkloads, ds.Name) {
+						log.InfoD("Deleting Workload Generating pods %v ", dep.Name)
+						err = pdslib.DeleteK8sDeployments(dep.Name, namespace)
+					} else if Contains(dataServicePodWorkloads, ds.Name) {
+						log.InfoD("Deleting Workload Generating pods %v ", pod.Name)
+						err = pdslib.DeleteK8sPods(pod.Name, namespace)
+					}
+					log.FailOnError(err, "error deleting workload generating pods")
+				})
+			}
+		})
+	})
+	JustAfterEach(func() {
+		defer EndTorpedoTest()
+
+		defer func() {
+			if !isDeploymentsDeleted {
+				Step("Delete created deployments")
+				resp, err := pdslib.DeleteDeployment(deployment.GetId())
+				log.FailOnError(err, "Error while deleting data services")
+				dash.VerifyFatal(resp.StatusCode, http.StatusAccepted, "validating the status response")
```

Comment:
add deployment id in the error log

Reply:
Noted. This needs to be changed in every TC and not just this one. Will raise a separate PR for this one later that will take care of all TCs